main.js

```diff
@@ -0,0 +1,251 @@
+    const locationData = await fetchCurrentLocationData(location);
+    addNow(locationData);
+
+    const fiveDayData = await fetchFiveDayData(location);
+    const processedData = await processFiveDayData(fiveDayData);
+    addFiveDay(processedData);
+  } catch (error) {
+    console.error("Error processing data:", error);
+  }
+}
+
+addCurrentLocation();
+
+async function getNow() {
+  try {
+    const nowData = await fetchNow();
+    addNow(nowData);
+  } catch (error) {
+    console.error("Error processing data:", error);
+  }
+}
+
+async function fetchNow() {
+  const query = document.querySelector("#query").value.replace(/\s+/g, "%20");
+
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${query}$&limit=1&appid=${apiKey}&units=${units}`;
+
+  const fetchedData = await fetch(url, {
+    method: "Get",
+    dataType: "json",
+  });
+
+  const nowData = await fetchedData.json();
+
+  return nowData;
+}
+
+async function addNow(data) {
+  const nowSection = document.querySelector("#now");
+  nowSection.replaceChildren();
+
+  const template = `      
+    <div class="row justify-content-center forecast-today my-3" id="now">
+      <div class="col-md-6">
+        <h3 class="text-center">${data.main.temp.toFixed(1)}°</h3>
+        <h4 class="text-center">${data.name}</h4>
+        <h5 class="text-center">${data.weather[0].main}</h5>
+      </div>
+      <div class="col-md-6">
+        <img
+          class="img-fluid mx-auto d-block"
+          src="https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png" )
+          alt="unable to load image"
+        />
+      </div>
+    </div>
+  `;
+
+  nowSection.insertAdjacentHTML("beforeend", template);
+}
+
+async function getFiveDay() {
+  try {
+    const coordinates = await fetchCoordinates();
+    const fiveDayData = await fetchFiveDayData(coordinates);
+    const processedData = await processFiveDayData(fiveDayData);
+    addFiveDay(processedData);
+  } catch (error) {
+    console.error("Error processing data:", error);
+  }
+}
+
+async function fetchCoordinates() {
+  const query = document.querySelector("#query").value.replace(/\s+/g, "%20");
+
+  const coordinatesCall = await fetch(
+    `https://api.openweathermap.org/geo/1.0/direct?q=${query}&limit=1&appid=${apiKey}
+    `,
+    {
+      method: "GET",
+      dataType: "json",
+    }
+  );
+
+  const coordinatesArray = await coordinatesCall.json();
+  const coordinates = coordinatesArray[0];
+
+  return coordinates;
+}
+
+async function fetchFiveDayData(coordinates) {
+  const fiveDayURL = `https://api.openweathermap.org/data/2.5/forecast?lat=${coordinates.lat}&lon=${coordinates.lon}&appid=${apiKey}&units=${units}`;
+
+  const fetchFiveDay = await fetch(fiveDayURL, {
+    method: "GET",
+    dataType: "json",
+  });
+
+  const fiveDayData = await fetchFiveDay.json();
+
+  return fiveDayData;
+}
+
+async function processFiveDayData(data) {
+  const dividedDayData = [];
+
+  function processDayData(day) {
```

Comment:
this function is too big, needs to be spread into smaller functions to help with readability and maintainability